src/main/java/control/EditMenuController.java

```diff
@@ -0,0 +1,276 @@
+package control;
+
+import java.util.LinkedList;
+
+import data.Camera;
+import data.CameraShot;
+import data.CameraTimeline;
+import data.CameraType;
+import data.DirectorTimeline;
+import data.ScriptingProject;
+import gui.modal.AddCameraModalView;
+import gui.modal.AddCameraTypeModalView;
+import gui.modal.EditProjectModalView;
+import gui.root.RootCenterArea;
+import javafx.scene.control.Label;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.HBox;
+import javafx.scene.paint.Color;
+
+public class EditMenuController {
+    
+    private ControllerManager controllerManager;
+    
+    private EditProjectModalView editModal;
+    private AddCameraTypeModalView typeModal;
+    private AddCameraModalView cameraModal;
+    
+    /**
+     * Construct a new EditMenuController.
+     * @param manager the ControllerManager that controls this EditMenuController
+     */
+    public EditMenuController(ControllerManager manager) {
+        this.controllerManager = manager;
+    }
+    
+    /**
+     * Start the edit project modal.
+     */
+    public void editProject() {
+        editModal = new EditProjectModalView(controllerManager.getRootPane());
+        editModal.getAddCameraButton().setOnMouseClicked(this::addCamera);
+        editModal.getDeleteCameraButton().setOnMouseClicked(this::deleteCamera);
+        editModal.getAddCameraTypeButton().setOnMouseClicked(this::addCameraType);
+        editModal.getDeleteCameraTypeButton().setOnMouseClicked(this::deleteCameraType);
+        editModal.getCancelButton().setOnMouseClicked(this::cancel);
+        editModal.getSaveButton().setOnMouseClicked(this::save);
+    }
+    
+    /**
+     * Handler for adding a camera.
+     * @param event the MouseEvent for this handler
+     */
+    private void addCamera(MouseEvent event) {
+        cameraModal = new AddCameraModalView(controllerManager.getRootPane(), editModal.getTypes());
+        cameraModal.getAddCameraButton().setOnMouseClicked(this::cameraAdded);
+    }
+    
+    /**
+     * Handler for when a camera is added.
+     * @param event the MouseEvent for this handler
+     */
+    private void cameraAdded(MouseEvent event) {
+        if (validateCameraData()) {
+            cameraModal.hideModal();
+            String name = cameraModal.getNameField().getText();
+            String description = cameraModal.getDescriptionField().getText();
+            int selectedIndex = cameraModal.getCameraTypes().getSelectionModel().getSelectedIndex();
+            CameraType type = cameraModal.getCameraTypeList().get(selectedIndex);
```

Comment:
long method with no whitespace, could you add some comments or whitespace to make it more readable